Formula/odo-dev.rb

```diff
@@ -0,0 +1,34 @@
+class OdoDev < Formula
+  desc "Developer-focused CLI for Kubernetes and OpenShift"
+  homepage "https://odo.dev"
+
+  url "https://github.com/redhat-developer/odo.git",
+    tag:      "v2.5.0",
+    revision: "724f16e689545dd4a81671da3e116a33df4832d3"
+
+  license "Apache-2.0"
+  head "https://github.com/redhat-developer/odo.git", branch: "main"
+
+  depends_on "go" => :build
+  conflicts_with "odo", because: "odo also ships 'odo' binary"
+
+  def install
+    system "make", "bin"
+    bin.install "odo"
+  end
+
+  test do
+    # try set preference
+    ENV["GLOBALODOCONFIG"] = "#{testpath}/preference.yaml"
+    shell_output("#{bin}/odo preference set ConsentTelemetry false")
+    assert_predicate testpath/"preference.yaml", :exist?
+
+    # test version
+    version_output = shell_output("#{bin}/odo version --client 2>&1").strip
+    tag = stable.instance_variable_get(:@resource).instance_variable_get(:@specs)[:tag]
```

Comment:
Is this not just `version`?

Reply:
`odo version` will try to access Kubernetes cluster to get also Kubernetes version, it will take a few seconds before it times out. `--client` flag is there to skip checking Kubernetes version.

Reply:
No, what I meant is: `tag == version` (modulo type conversions)

Reply:
I'm sorry. But don't know what you mean :-/ 
Wouldn't be enough if I just check the version. For example something like this `assert_match(/odo v#{version} \([a-f0-9]{9}\)/, version_output)`

Reply:
I mean that we don't need to reach into the internals of the formula to define `tag` when we can just use the `version` variable instead.

> Wouldn't be enough if I just check the version. For example something like this `assert_match(/odo v#{version} \([a-f0-9]{9}\)/, version_output)`

This is exactly what I was suggesting here: https://github.com/Homebrew/homebrew-core/pull/93846#discussion_r792618695

Reply:
ok. did that https://github.com/Homebrew/homebrew-core/pull/93846/commits/ac052ef82c1776ef9d85d13fdd5b66d0e46af55e#diff-62d004c8cf3cfdc05e6facbca1421a8ade64bbb41a9e3882254107cba4e30158R25